doc/source/ray-core/tasks/fault-tolerance.rst

```diff
@@ -42,8 +48,89 @@ You can experiment with this behavior by running the following code.
         except ray.exceptions.WorkerCrashedError:
             print('FAILURE')
 
+You can also control whether application-level errors are retried, and even **which**
+application-level errors are retried, via the ``retry_exceptions`` argument. This is
+``False`` by default, so if your application code within the Ray task raises an
+exception, this failure will **not** be retried. This is to ensure that Ray is not
+retrying non-idempotent tasks when they have partially executed.
+However, if your tasks are idempotent, then you can enable application-level error
+retries with ``retry_exceptions=True``, or even retry a specific set of
+application-level errors (such as a class of exception types that you know to be
+transient) by providing an allowlist of exceptions:
+
+.. code-block:: python
+
+    import numpy as np
```

Comment:
Can you pull this out into a literalinclude block of doccode?

Reply:
Sure, are we doing this now as a PR blocker whenever we touch any Ray Core feature guides? Because I think that most of the Ray Core feature guides have these inline code blocks.

Reply:
We are planning to do a ray core doc revamp to fix the issue. But for new code, I think it's better to start with new standard.